src/femtomes.cpp

```diff
@@ -103,17 +120,97 @@ int GPSDriverFemto::handleMessage(int len)
 		_gps_position->vel_ned_valid = _femto_uav_gps.vel_ned_valid;
 		_gps_position->satellites_used = _femto_uav_gps.satellites_used;
 
-		heading = _femto_uav_gps.heading - _heading_offset;
-
-		if (heading > M_PI_F) {
-			heading -= 2.f * M_PI_F;// final range is [-pi, pi]
-		}
+        if(_femto_uav_gps.heading_type == 6)
+        {
+            float heading = _femto_uav_gps.heading;
+            heading *= M_PI_F / 180.0f; // deg to rad, now in range [0, 2pi]
+            heading -= _heading_offset; // range: [-pi, 3pi]
+
+            if (heading > M_PI_F)
+            {
+                heading -= 2.f * M_PI_F; // final range is [-pi, pi]
+            }
+            _gps_position->heading = heading;
+        }
+        else
+        {
+            _gps_position->heading = NAN;
+        }
 
-		_gps_position->heading = heading;
 		_gps_position->timestamp = gps_absolute_time();
 
 		ret = 1;
 	}
+	else if(messageid == FEMTO_MSG_ID_RTCM3)    /**< rtcm3 */
+    {
+        gotRTCMMessage(_rtcm_parsing->message(), _rtcm_parsing->messageLength());
+        decodeInit();
+        ret = 1;
+    }
+	else if(messageid == FEMTO_MSG_ID_GPGGA && (memcmp(_femto_msg.data + 3, "GGA,", 3) == 0)) /**< GPGGA */
```

Comment:
Is this only ever received by the base station?
If it's possible to receive this on the vehicle side as well, `activateRTCMOutput` should not be called (you can add a check for `_output_mode`).

Reply:
Thank you ! You are right, I should add a check for _output_mode here. Although it's only for base stations